lib/reaktor/notification/notifier.rb

```diff
@@ -1,56 +1,26 @@
-require 'observer'
 require 'singleton'
+require 'slack-notifier'
+require 'yaml'
 
 module Reaktor
 module Notification
   class Notifier
     include Singleton
-    include Observable
 
     attr_accessor :notification
 
     def initialize
-      # if user wants to override where to look for custom notifiers
-      @notifier_dir = ENV['REAKTOR_NOTIFIERS_DIR']
-      # if no custom notifier dir found, use default
-      current_dir = Dir.pwd
-      @notifier_dir ||= "#{current_dir}/lib/reaktor/notification/active_notifiers"
-      loadNotifiersAsObservers(@notifier_dir)
-    end
-
-    # load all notifiers and add them as observers
-    # @param notifier_dir - directory where notifier ruby files live
-    def loadNotifiersAsObservers(notifier_dir)
-      # dynamically load each notifier found
-      Dir.glob("#{notifier_dir}/**/*.rb").each{|f| require f}
-      @notification = nil
+      @config = YAML.load_file(('config/notifiers.yml'))
       @logger ||= Logger.new(STDOUT, Logger::INFO)
-      @logger.info("NOTIFIER_DIR = #{notifier_dir}")
-      Dir.glob("#{notifier_dir}/*.rb").each do |f|
-        @logger.debug("loading notifier from file: #{f}")
-        clazz = getClassFromFile(f)
-        self.add_observer(clazz.instance)
-      end
     end
 
-    # get the class object from the file it's declared in
-    # @param filename - the ruby class file
-    # returns the class object name
-    def getClassFromFile(filename)
-      notifier_base_name = File.basename(filename, ".rb")
-      instance_name = notifier_base_name.capitalize
-      # split on "::" and iterate the results, returning the last element found,
-      # which is the class name we're after
-      "Notifiers::#{instance_name}".split('::').inject(Object) {|o,c| o.const_get c}
+    def send_message(message,room='default')
+      webhook_uri = @config['notifiers'][room]
```

Comment:
If we keep the YAML structure, please add error handling to this in case we reference a room incorrectly and the `webhook_uri` is nil